app/models/legal_aid_application.rb

```diff
@@ -10,6 +10,19 @@ class LegalAidApplication < ApplicationRecord
 
   validate :proceeding_type_codes_existence
 
+  def self.find_by_secure_id!(secure_id)
+    secure_data = SecureData.for(secure_id)
+    find_by! secure_data[:legal_aid_application]
+  end
+
+  def generate_secure_id
+    SecureData.create_and_store!(
+      legal_aid_application: { id: id },
+      # Timestamp so token unique for each secure id
+      timestamp: Time.now.to_s
```

Comment:
If the only goal of this is to generate a unique hash, we could use `SecureRandom.hex`. Because `Time.now.to_s` is only unique per second.